src/main/java/com/redhat/devtools/lsp4ij/LanguageServerFactory.java

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij;
+
+import com.intellij.openapi.project.Project;
+import com.redhat.devtools.lsp4ij.client.LanguageClientImpl;
+import com.redhat.devtools.lsp4ij.server.StreamConnectionProvider;
+import org.eclipse.lsp4j.services.LanguageServer;
+
+/**
+ * Language server factory API.
+ */
+public interface LanguageServerFactory {
+
+    StreamConnectionProvider createConnectionProvider(Project project);

```

Comment:
you should add @NotNull / @Nullable annotations on the methods params / return types

Reply:
also, javadoc